main.cpp

```diff
@@ -228,346 +305,521 @@ struct University
-    //toggle ON and OFF
-    void toggleONOFF(bool LFO = true);
-    //change the interaction time between signals
-    void changeSignalsInteraction(float amount, int phaseOffset);
+    double modulateSignal(double x, 
+                          std::string routeAssign = "Oscillator");
+    bool toggleEnablement(bool shouldBeOn = true);  
+    double changeSignalsInteraction(double x, 
+                                    double amount, 
+                                    double phaseOffset);
 };
 
-/*
-Thing 7) oscillator
-5 properties:
-    1) frequency (float)
-    2) finetune (float)
-    3) waveform shape (std::string)  
-    4) pulse width (float) 
-    5) octave (int)  
-3 things it can do:
-    1) generate audio signals
-    2) load samples from ROM
-    3) playback samples from ROM
- */
+double LFO::modulateSignal(double x, std::string routeAssign)
+{
+    if (routeAssign == "Oscillator")
+    {
+        double y = 1;
+        x = x + y;
+    }
+    return x; //This is a dummy example
+}
+
+bool LFO::toggleEnablement(bool shouldBeOn)
+{
+    if (shouldBeOn == false)
+    {
+        bypassState = true;
+    }
+    return bypassState; //This is a dummy example
+}
+
+double LFO::changeSignalsInteraction(double x, double , double )
+{
+    int fs = 44'100;
+    for (int i = 0; i < fs; i++)
+    {
+        x = amount - phaseOffset;
+    }
+    return x; //This is a dummy example
+}
 
 struct Oscillator
 {
-    //frequency [Hz]
-    float frequency = 440; 
-    //Frequency fine tune variation
+    double frequency = 440; 
     float finetune = 0.01f; 
-    //waveform shape (Signal shape)
     std::string waveformShape = "Square"; 
-    //pulse width (% of the positive/negative signal amplitude to be wider or smaller )
     float pulseWidth = 0.3f; 
-    //octave (n times the original frequency )
     int octave = 2; 
+    double loadedROM;
+    double storedROM;
+    bool playButton;
 
-    //generate audio signals
-    void generateSignal(Oscillator Square);
-    //load samples from ROM
-    void loadROMSamples(std::string selectStorageDevice = "SD",
+    double generateSignal(Oscillator Square, double x); 
+    double loadROMSamples(std::string selectStorageDevice = "SD",
                         bool isAudioFormat = true);
-    //playback samples from ROM
-    void playbackROMSamples(bool anyKeyPressed = false);
+    void playbackROMSamples(double x, bool anyKeyPressed = false);
 };
 
-/*
-Thing 8) filters
-5 properties:
-    1) gain (float)
-    2) bandwidth (float)
-    3) frequency (float)
-    4) type (std::string)
-    5) drive (float)
-3 things it can do:
-    1) boost or cut frequencies on a signal
-    2) overDrive the signal
-    3) give a better sonority to an instrument 
- */
+double Oscillator::generateSignal(Oscillator Square, double x)
+{
+    if (Square.waveformShape == "Square")
+    {
+        x = 0.5 * (frequency); //This is a dummy example
+    }
+    return x; 
```

Comment:
Don't pass in an instance of the `Oscillator` class into an Oscillator member function!
If you're trying to use `waveformShape`, which is a member variable of the `Oscillator` instance being used when this function is called, just use it directly inside the function.   
```
if( waveformShape == "Square" )
    x = 0.5 * frequency;

return x;
```
- [ ] revise